aten/src/ATen/native/cuda/BinaryOpsKernel.cu

```diff
@@ -67,13 +67,22 @@ static void div_kernel_cuda(TensorIterator& iter) {
 
 template <typename scalar_t>
 void mul_kernel_impl(TensorIterator& iter) {
-  gpu_binary_kernel(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-    return a * b;
-  });
+  // Some of the compilers that we support give an error if we try multiplying two
```

Comment:
is it actually an error or is it a warning and we have Werror on?  In general it would be good to give an example, for example, "on gcc 7.4 this gives XXX".